### PR TITLE
feat: [AAP-19417] Enable to send vault credentials to rulebook

### PR DESCRIPTION
**Pull request by hsong-rh**
This PR adds the logic to send vault-type credentials to the rulebook engine via WebSocket channels

Resolves: [AAP-19417](https://issues.redhat.com/browse/AAP-19417)

### CODE REVIEW

File: src/aap_eda/wsapi/consumers.py [hunk 1]
```diff
@@ -121,6 +124,13 @@ async def handle_workers(self, message: WorkerMessage):
                 ssl_verify=settings.EDA_CONTROLLER_SSL_VERIFY,
             )
             await self.send(text_data=controller_info.json())
+
+        vault_data = await self.get_vault_passwords(message)
+        if vault_data:
+            vault_collection = VaultCollection(data=vault_data)
+            logger.error(vault_collection.json())
```
Comment (mkanoor) [inline src/aap_eda/wsapi/consumers.py:131]: @hsong-rh This line should be removed it will leak passwords

File: src/aap_eda/wsapi/consumers.py [hunk 2]
```diff
@@ -208,9 +218,12 @@ def insert_event_related_data(self, message: AnsibleEventMessage) -> None:
 
     @database_sync_to_async
     def insert_audit_rule_data(self, message: ActionMessage) -> None:
-        job_id = message.job_id
-        job_instance = models.JobInstance.objects.filter(uuid=job_id).first()
-        job_instance_id = job_instance.id if job_instance else None
+        job_instance_id = None
+        if message.job_id:
```
Comment (mkanoor) [inline src/aap_eda/wsapi/consumers.py:222]: @hsong-rh I think this should be in a separate PR. I think the newer versions of pydantic does the uuid validation I am not sure if the version we are using is doing the validation.

File: src/aap_eda/wsapi/consumers.py [hunk 3]
```diff
@@ -330,3 +343,35 @@ def get_awx_token(self, message: WorkerMessage) -> tp.Optional[str]:
...
+                )
+            )
+
+        if activation.system_vault_credential:
+            password = (
+                activation.system_vault_credential.secret.get_secret_value()
+            )
+            vault_passwords.append(
+                VaultPassword(
+                    label=activation.system_vault_credential.key,
```
Comment (mkanoor) [inline src/aap_eda/wsapi/consumers.py:373]: @hsong-rh Is this key or vault_credential?

File: tests/integration/wsapi/test_consumer.py [hunk 4]
```diff
@@ -9,6 +9,7 @@
 from pydantic.error_wrappers import ValidationError
 
 from aap_eda.core import models
+from aap_eda.core.enums import CredentialType
```
Comment (mkanoor) [inline tests/integration/wsapi/test_consumer.py:12]: @hsong-rh can we add a test with system vault credential. I am hoping the label gets set when we add it in.

File: src/aap_eda/wsapi/messages.py [hunk 5]
```diff
@@ -85,6 +85,17 @@ class ControllerInfo(BaseModel):
     ssl_verify: str
 
 
+class VaultPassword(BaseModel):
+    type: str = "VaultPassword"
```
Comment (Alex-Izquierdo) [inline src/aap_eda/wsapi/messages.py:89]: - What is the purpose of a type str? I think it should be a enum: https://docs.pydantic.dev/2.3/usage/types/enums/
- Do we really need a type? What are the other types? because the field "password" is a mandatory string.

File: src/aap_eda/wsapi/messages.py [hunk 6]
```diff
@@ -38,14 +38,14 @@ class ActionMessage(Message):
...
-    message: Optional[str]
-    kind: Optional[str]
+    playbook_name: Optional[str] = ""
+    job_template_name: Optional[str] = ""
+    organization: Optional[str] = ""
+    job_id: Optional[str] = ""
+    rc: Optional[int] = 0
+    delay: Optional[float] = 0.0
+    message: Optional[str] = ""
+    kind: Optional[str] = ""
```
Comment (Alex-Izquierdo) [inline src/aap_eda/wsapi/messages.py:48]: Why do we need empty default values when the type is already optional?

File: tests/integration/wsapi/test_consumer.py (hunk 4 above)
Comment (hsong-rh) [reply tests/integration/wsapi/test_consumer.py:12]: Will add it later when db migrations are merged

File: src/aap_eda/wsapi/messages.py (hunk 6 above)
Comment (hsong-rh) [reply src/aap_eda/wsapi/messages.py:48]: They will be required when pydantic package upgrades

File: src/aap_eda/wsapi/messages.py (hunk 5 above)
Comment (mkanoor) [reply src/aap_eda/wsapi/messages.py:89]: @Alex-Izquierdo Every message sent on the WebSocket Channel has a type they are all listed [here](https://github.com/ansible/eda-server/blob/eed9db3923c1c2b1674fdddd335142beb55d4768/src/aap_eda/wsapi/messages.py#L21)

File: tests/integration/wsapi/test_consumer.py (hunk 4 above)
Comment (hsong-rh) [reply tests/integration/wsapi/test_consumer.py:13]: @mkanoor tests are added

**Review by mkanoor — APPROVED** [review]

File: src/aap_eda/wsapi/messages.py (hunk 6 above)
Comment (hsong-rh) [reply src/aap_eda/wsapi/messages.py]: Removed the changes required by pydantic 2.5.

File: tests/integration/wsapi/test_consumer.py [hunk 7]
```diff
@@ -427,6 +523,172 @@ def get_job_instance_event_count():
     return models.JobInstanceEvent.objects.count()
 
 
+@database_sync_to_async
+def _prepare_activation_instance_with_vault_credentials():
```
Comment (bzwei) [inline tests/integration/wsapi/test_consumer.py:527]: can you consolidate three _prepare functions into one with controlling parameters?
Similarly you may be able to combine three tests into one parameterized test.

**Review by Alex-Izquierdo — APPROVED** [review]

**Comment by bzwei** [comment]
This PR has dependency on https://github.com/ansible/ansible-rulebook/pull/651, which has been merged.

File: tests/integration/wsapi/test_consumer.py (hunk 7 above)
Comment (hsong-rh) [reply tests/integration/wsapi/test_consumer.py:527]: These are the async function tests, each test case is marked as `transaction=True`. I don't think we can separate them as  parameterized tests.

**Review by Alex-Izquierdo — APPROVED** [review]
